File: src/py_package/py_package/robot_news_station_listener.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class RobotNewsStationListener(Node):
    def __init__(self):
        super().__init__("robot_news_station_listener")
        self.subscriber_ = self.create_subscription(String, "robot_news", self.callback_robot_news, 10)
    
    def callback_robot_news(self, msg: String):
        self.get_logger().info(f"Hi C3P0, I hear you: {msg.data}")

def main(args=None):
    rclpy.init(args=args)

    node = RobotNewsStationListener()

    rclpy.spin(node)

    rclpy.shutdown()


if __name__ == "__main__":
    main()       

File: src/cpp_package/src/robot_news_station_listener.cpp
#include "cpp_package/robot_news_station_listener.hpp"
#include <rclcpp/rclcpp.hpp>

RobotNewsStationListener::RobotNewsStationListener() : Node("robot_news_station_listener")
{
    listener_ = this->create_subscription<example_interfaces::msg::String>("robot_news", 10, std::bind(&RobotNewsStationListener::ReadNews, this, std::placeholders::_1));
    RCLCPP_INFO(this->get_logger(), "Robot news station listener");
}

void RobotNewsStationListener::ReadNews(const example_interfaces::msg::String::SharedPtr msg)
{
    RCLCPP_INFO(this->get_logger(), msg->data.c_str());
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<RobotNewsStationListener>();

    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}